src/index.js

```diff
@@ -94,6 +94,7 @@ export default class MediaQuery extends React.Component {
     const wrapChildren = this.props.component ||
       childrenCount > 1 ||
       typeof this.props.children === 'string' ||
+      Array.isArray(this.props.children) && childrenCount == 1 ||
```

Comment:
yes, [this test](https://github.com/contra/react-responsive/blob/master/test/index_test.js#L130) fails without this check.